src/core/ddsc/src/dds_qos.c

```diff
@@ -618,8 +618,20 @@ bool dds_qget_partition (const dds_qos_t * __restrict qos, uint32_t *n, char ***
     *n = qos->partition.n;
```

Comment:
I don't like the behaviour that, for an entity created with an empty set of partitions:
```
dds_qget_partition (q, &nA, NULL);
dds_qget_partition (q, &nB, &ps);
assert (nA == nB);
```
will fail.